lib/route_consistency_checker.rb

```diff
@@ -0,0 +1,145 @@
+require "set"
+
+require "gds_api/router"
+
+class RouteConsistencyChecker
+  attr_reader :errors
+
+  def initialize(routes, router_data)
+    @routes = load_routes(routes)
+    @remaining_routes = Set.new(@routes.keys)
+    @errors = Hash.new { |hash, key| hash[key] = [] }
+    @router_data = load_router_data(router_data)
+  end
+
+  def check_content
+    content_items_to_check.each do |content_item|
```

Comment:
maybe see if we can memoize things that are checked here so they can be subtracted from routes to check? to try make remaining routes usage more explicit